From 50541cbb660c0f651a4178f1c1996540209534d5 Mon Sep 17 00:00:00 2001
From: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
Date: Wed, 13 Sep 2023 00:18:02 +0800
Subject: [PATCH 27/35] arm64: dts: add dma-coherent property to rpmsg-ca53
 node

Add the dma-coherent to the rpmsg-ca53 device to make vring
memory and rpmsg buffer using cacheable memory.

Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
---
 arch/arm64/boot/dts/freescale/imx8m-rpmsg-ca53.dtsi  | 1 +
 arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi | 1 +
 arch/arm64/boot/dts/freescale/imx93-rpmsg-ca55.dtsi  | 1 +
 3 files changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8m-rpmsg-ca53.dtsi b/arch/arm64/boot/dts/freescale/imx8m-rpmsg-ca53.dtsi
index 95d56dd208e0..77f0d92ceb69 100644
--- a/arch/arm64/boot/dts/freescale/imx8m-rpmsg-ca53.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8m-rpmsg-ca53.dtsi
@@ -26,6 +26,7 @@ vdevbuffer_ca53: vdevbuffer-ca53@b8700000 {
 	rpmsg-ca53 {
 		compatible = "fsl,imx8mm-rpmsg";
 		reg = <0x0 0xb8600000 0x0 0x10000>; /* 64K for one rpmsg instance */
+		dma-coherent;
 		memory-region = <&vdevbuffer_ca53>;
 		mbox-names = "tx", "rx", "rxdb";
 		mboxes = <&gen_sw_mbox 0 0
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi
index dcc4afb47300..1bb2a8602379 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp-rpmsg-ca53.dtsi
@@ -26,6 +26,7 @@ vdevbuffer_ca53: vdevbuffer-ca53@fe200000 {
 	rpmsg-ca53 {
 		compatible = "fsl,imx8mm-rpmsg";
 		reg = <0x0 0xfe100000 0x0 0x10000>; /* 64K for one rpmsg instance */
+		dma-coherent;
 		memory-region = <&vdevbuffer_ca53>;
 		mbox-names = "tx", "rx", "rxdb";
 		mboxes = <&gen_sw_mbox 0 0
diff --git a/arch/arm64/boot/dts/freescale/imx93-rpmsg-ca55.dtsi b/arch/arm64/boot/dts/freescale/imx93-rpmsg-ca55.dtsi
index 499543f34258..a9780af05d71 100644
--- a/arch/arm64/boot/dts/freescale/imx93-rpmsg-ca55.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx93-rpmsg-ca55.dtsi
@@ -26,6 +26,7 @@ vdevbuffer_ca55: vdevbuffer-ca55@fe200000 {
 	rpmsg-ca55 {
 		compatible = "fsl,imx8mm-rpmsg";
 		reg = <0x0 0xfe100000 0x0 0x10000>; /* 64K for one rpmsg instance */
+		dma-coherent;
 		memory-region = <&vdevbuffer_ca55>;
 		mbox-names = "tx", "rx", "rxdb";
 		mboxes = <&gen_sw_mbox 0 0
-- 
2.34.1

